[azure-core-cpp, azure-identity-cpp] Update to latest version

azure-sdk opened the pull request:
Update vcpkg ports for Azure SDK release. This release may contain multiple ports.

azure-sdk committed:
[azure-core-cpp] Update to 1.0.0-beta.8
## 1.0.0-beta.8 (2021-04-07)

### New Features

- Added `Azure::Core::Url::GetScheme()`.
- Added `Azure::Core::Context::TryGetValue()`.
- Added `Azure::Core::Context::GetDeadline()`.
- Added `Azure::Core::Credentials::TokenCredentialOptions`.
- Added useful fields to the `Azure::Core::RequestFailedException` class such as `StatusCode`, `ReasonPhrase`, and the `RawResponse`, for better diagnosis of errors.

### Breaking Changes

- Simplified the `Response<T>` API surface to expose two public fields with direct access: `T Value` and a `unique_ptr` to an `Azure::Core::Http::RawResponse`.
- Renamed `Azure::Nullable<T>::GetValue()` to `Value()`.
- Removed from `Azure::Core::Http::Request`:
  - `SetUploadChunkSize()`.
  - `GetHTTPMessagePreBody()`.
  - `GetUploadChunkSize()`.
  - `GetHeadersAsString()`.
- Changes to `Azure::Core::Http::RawResponse`:
  - Removed `SetHeader(std::string const& header)`
  - Removed `SetHeader(uint8_t const* const first, uint8_t const* const last)`.
  - Removed `GetMajorVersion()`.
  - Removed `GetMinorVersion()`.
  - Renamed `GetBodyStream()` to `ExtractBodyStream()`.
- Changes to `Azure::Core::Context`:
  - Removed `Get()` and `HasKey()` in favor of a new method `TryGetValue()`.
  - Changed input parameter type of `WithDeadline()` to `Azure::DateTime`.
- Removed `Azure::Core::PackageVersion`.
- Removed from `Azure::Core::Http::Policies` namespace: `HttpPolicyOrder`, `TransportPolicy`, `RetryPolicy`, `RequestIdPolicy`, `TelemetryPolicy`, `BearerTokenAuthenticationPolicy`, `LogPolicy`.
- Removed `AppendQueryParameters()`, `GetUrlWithoutQuery()` and `GetUrlAuthorityWithScheme()` from `Azure::Core::Url`.
- Changed the `Azure::Core::Http::HttpMethod` regular enum into an extensible enum class and removed the `HttpMethodToString()` helper method.
- Introduced `Azure::Core::Context::Key` class which takes place of `std::string` used for `Azure::Core::Context` keys previously.
- Changed the casing of `SSL` in API names to `Ssl`:
  - Renamed type `Azure::Core::Http::CurlTransportSSLOptions` to `CurlTransportSslOptions`.
  - Renamed member `Azure::Core::Http::CurlTransportOptions::SSLOptions` to `SslOptions`.
  - Renamed member `Azure::Core::Http::CurlTransportOptions::SSLVerifyPeer` to `SslVerifyPeer`.

### Other changes and Improvements

- Moved `Azure::Core::Http::Request` to its own header file from `http.hpp` to `inc/azure/core/http/raw_response.hpp`.
- Moved `Azure::Core::Http::HttpStatusCode` to its own header file from `http.hpp` to `inc/azure/core/http/http_status_code.hpp`.

azure-sdk committed:
[azure-identity-cpp] Update to 1.0.0-beta.5
## 1.0.0-beta.5 (2021-04-07)

### New Features

- Add Active Directory Federation Service (ADFS) support to `ClientSecretCredential`.

### Breaking Changes

- Removed `Azure::Identity::PackageVersion`.

ahsonkhan reviewed (approved):
@Jinming-Hu use the same branch name (and PR name) for the storage package releases

autoantwort commented:
So this should be merged into vcpkg and is not spam?

Jinming-Hu commented:
We should create this kind of PR as draft next time.

NancyLi1013 commented:
Hi @ahsonkhan 

Thanks for your PR.

`azure-storage-common-cpp` failed with the following errors:
```
D:\buildtrees\azure-storage-common-cpp\src\0.0-beta.9-fd2c305c02.clean\sdk\storage\azure-storage-common\src\storage_credential.cpp(99): error C2248: 'Azure::Core::Url::AppendQueryParameters': cannot access private member declared in class 'Azure::Core::Url'
D:\installed\x86-windows\include\azure/core/url.hpp(67): note: see declaration of 'Azure::Core::Url::AppendQueryParameters'
D:\installed\x86-windows\include\azure/core/url.hpp(46): note: see declaration of 'Azure::Core::Url'
```
Could you please look into this?

ahsonkhan commented:
> So this should be merged into vcpkg and is not spam?

Yes, it is intended to be merged. We will no longer open non-draft PRs that are just for tests, and minimize/limit even the draft PRs that are open for testing (ideally just one - https://github.com/microsoft/vcpkg/pull/17119).

> We should create this kind of PR as draft next time.

Why? We intend to publish the updated versions to the packages and merge this PR.

ahsonkhan commented:
> Could you please look into this?

Thanks @NancyLi1013 - we will add a commit to this PR to update the versions of the azure-storage-* packages, that are built on top of the changes to azure-core-cpp, this week and that should address the the error.

ahsonkhan commented:
![image](https://user-images.githubusercontent.com/6527137/113965498-56d2f100-97e2-11eb-865b-c7dc031dff32.png)

@NancyLi1013 can you help me understand what format you prefer the PR titles to be named as? The PR title is generated and our CI/CD infra relies on it when adding new commits. If you let us know what format you prefer the name to be, we can setup the PRs to always be opened with the expected name and avoid having the title be renamed.

Its possible that a PR title rename after the PR is opened causes some issues in our infra (I will find out from our infra team).

Just wanted to give you a heads up and understand the rationale for the title change. We may need to leave PR titles as they were, in the future.

NancyLi1013 commented:
@ahsonkhan Thanks for your so detailed information about PR titles.

Generally, we prefer to rename the titles as follows:

- For new ports, the title like this `[PortName] Add new port`
- For update ports, the title like this `[PortName] Update to ***(version or commits)`
- For bugs or improvement, the title like this `[PortName] Fix ***(describe the issue simply)`
- For vcpkg issue, the title like this `[vcpkg] ***(describe what issue you will fix simply)`

But currently, there is no document about PR titles.

ahsonkhan commented:
FYI @RickWinter

azure-sdk committed:
[azure-storage-common-cpp] Update to 12.0.0-beta.10
## 12.0.0-beta.10 (2021-04-16)

### New Features

- Added server timeout support.
- Added `PagedResponse<T>` for returning paginated collections.

### Breaking Changes

- Removed `Azure::Storage::Common::PackageVersion`.
- Moved `ReliableStream` to internal namespace.
- Removed `HttpGetterInfo` and `HTTPGetter` from the `Azure::Storage` namespace.

azure-sdk committed:
[azure-storage-blobs-cpp] Update to 12.0.0-beta.10
## 12.0.0-beta.10 (2021-04-16)

### Breaking Changes

- Removed `Azure::Storage::Blobs::PackageVersion`.
- Renamed `GetUserDelegationKeyOptions::startsOn` to `StartsOn`.
- Replaced all paginated collection functions that have the SinglePage suffix with pageable functions returning a `PagedResponse<T>`-derived type. The options are also renamed accordingly.
  - `BlobServiceClient::ListBlobContainers()`.
  - `BlobServiceClient::FindBlobsByTags()`.
  - `BlobContainerClinet::ListBlobs()`.
  - `BlobContainerClient::ListBlobsByHierarchy()`.
  - `PageBlobClient::GetPageRanges()`.
  - `PageBlobClient::GetPageRangesDiff()`.
  - `PageBlobClient::GetManagedDiskPageRangesDiff()`.
- Renamed `FilterBlobItem` to `TaggedBlobItem`.
  - `FindBlobsByTags()` now returns `FindBlobsByTagsPagedResponse` and the field `FindBlobsByTagsSinglePageResult::Items` was renamed to `FindBlobsByTagsPagedResponse::TaggedBlobs`.

azure-sdk committed:
[azure-storage-files-datalake-cpp] Update to 12.0.0-beta.10
## 12.0.0-beta.10 (2021-04-16)

### Breaking Changes

- Removed `Azure::Storage::Files::DataLake::PackageVersion`.
- Renamed `GetUserDelegationKeyOptions::startsOn` to `StartsOn`.
- Replaced all paginated collection functions that have the SinglePage suffix with pageable functions returning a `PagedResponse<T>`-derived type. The options are also renamed accordingly.
  - `DataLakeServiceClient::ListFileSystems()`.
  - `DataLakeFileSystemClient::ListPaths()`.
  - `DataLakeDirectoryClient::ListPaths()`.
- Removed `DataLakePathClient::SetAccessControlListRecursiveSinglePage()`, `UpdateAccessControlListRecursiveSinglePage()` and `RemoveAccessControlListRecursiveSinglePage()`.

### Bug Fixes

- Rename functions always fail because `/` was left out in the renamed source path.

azure-sdk committed:
[azure-storage-files-shares-cpp] Update to 12.0.0-beta.10
## 12.0.0-beta.10 (2021-04-16)

### Breaking Changes

- Removed `Azure::Storage::Files::Shares::PackageVersion`.
- Renamed `GetUserDelegationKeyOptions::startsOn` to `StartsOn`.
- Removed `ShareClient::ListFilesAndDirectories()`.
- Replaced all paginated collection functions that have the SinglePage suffix with pageable functions returning a `PagedResponse<T>`-derived type. The options are also renamed accordingly.
  - `ShareServiceClient::ListShares()`.
  - `ShareDirectoryClient::ListFilesAndDirectories()`.
  - `ShareDirectoryClient::ListHandles()`.
  - `ShareFileClient::ListHandles()`.
- Removed `ShareDirectoryClient::ForceCloseAllHandlesSinglePage()` and `ShareFileClient::ForceCloseAllHandlesSinglePage()`.

ahsonkhan committed:
Update vcpkg ports to use a manifest json file instead of a CONTROL file.

ahsonkhan committed:
Update git tree sha versions using the command 'x-add-version --all --overwrite-version'

ras0219-msft commented:
LGTM, thanks!
